src/cairoUtilFuncGen/abi/abiEncode.ts

```diff
@@ -42,23 +43,29 @@ export class AbiEncode extends AbiBase {
     this.memoryRead = memoryRead;
   }
 
-  public getOrCreate(types: TypeNode[]): string {
-    const key = types.map((t) => t.pp()).join(',');
-    const existing = this.generatedFunctions.get(key);
-    if (existing !== undefined) {
-      return existing.name;
-    }
-
-    const [params, encodings] = types.reduce(
-      ([params, encodings], type, index) => {
+  public getOrCreate(types: TypeNode[]): GeneratedFunctionInfo {
+    const [params, encodings, functionsCalled] = types.reduce(
+      ([params, encodings, functionsCalled], type, index) => {
         const cairoType = CairoType.fromSol(type, this.ast, TypeConversionContext.Ref);
         params.push({ name: `param${index}`, type: cairoType.toString() });
-        encodings.push(
-          this.generateEncodingCode(type, 'bytes_index', 'bytes_offset', '0', `param${index}`),
+
+        const [paramEncoding, paramFunctionsCalled] = this.generateEncodingCode(
+          type,
+          'bytes_index',
+          'bytes_offset',
+          '0',
+          `param${index}`,
         );
-        return [params, encodings];
+
+        encodings.push(paramEncoding);
+
+        return [params, encodings, functionsCalled.concat(paramFunctionsCalled)];
```

Comment:
We usually prefer destructuring over concat in the code afaik

Reply:
Yes, style varies from writer to writer. What we try to do is follow a functional programming style, with a new variable each iteration. I think some writers don't do that at all like this one (like params and encodings with push)

Reply:
Normalizing the style throught the whole writers can be a time consuming stuff. I would suggest delaying it, if benefits are not that big (like in this case in my opinion)